### ISSUE TITLE
Heading selection UX improvements

### ISSUE THREAD

**Issue by aaronshekey** [StackExchange/Stacks-Editor#105]
So glad to see headings landed! There's one slightly weird thing about the UX here.

Let's say you've got a Heading 2:

<img width="222" alt="image" src="https://user-images.githubusercontent.com/1369864/159578186-55e02e51-1044-43a1-97a5-c763317bc067.png">

When I click Heading 1, it toggles all headings off instead of switching the heading 2 to a heading 1. That interaction should select the Heading 1, not toggle off a heading 2. :v:

🤔

**Comment by b-kelly** [comment]
Good catch. I'd go so far as to call that a bug, rather than an enhancement. I started on some tests for this area to catch things like this, but apparently I should have finished them rather than leaving them in a TODO state...
